allennlp_models/generation/dataset_readers/seq2seq.py

```diff
@@ -91,16 +101,22 @@ def __init__(
             or target_add_start_token
             or target_add_end_token
         ):
+            # Check that the tokenizer correctly appends the start and end tokens to
+            # the sequence without splitting them.
+            tokens = self._source_tokenizer.tokenize(start_symbol + " " + end_symbol)
             try:
-                self._start_token, self._end_token = self._source_tokenizer.tokenize(
-                    start_symbol + " " + end_symbol
-                )
-            except ValueError:
+                start_token, end_token = tokens[0], tokens[-1]
+                if start_token.text != start_symbol or end_token.text != end_symbol:
+                    raise IndexError
```

Comment:
My only issue with this that it will produce a stack trace that is a little confusing.

How about something like this:

```python
err_msg = f"Bad start or end symbol ('{start_symbol}', '{end_symbol}') for tokenizer {self._source_tokenizer}"
try:
    start_token, end_token = tokens[0], tokens[-1]
except IndexError:
    raise ValueError(err_msg)
if start_token.text != start_symbol or end_token.text != end_symbol:
    raise ValueError(err_msg)
```

Reply:
Yup, good call this is better.